CODING_GUIDELINES.dox

```diff
@@ -0,0 +1,335 @@
+
+
+================================================================================
+\section code_guidelines_4 Control statements
+
+Insert new line before
+
+- else in an if statement
+- catch in a try statement
+- while in a do statement
+
+
+--------------------------------------------------------------------------------
+\subsection code_guidelines_4_1 if else
+
+- put then statement, return or throw to new line
+- keep else if on one line 
+
+~~~~~~~~~~~~~
+if (true)
+  return;
+
+if (true)
+{
+  [...]
+} 
+else if (false)
+{
+  return;
+} 
+else
+  return;
+~~~~~~~~~~~~~
+
+--------------------------------------------------------------------------------
+\subsection code_guidelines_4_2 switch / case
+
+~~~~~~~~~~~~~[.cpp}
+switch (cmd)
+{
+  case x:
+  {
+    doSomething();
+    break;
+  }
+  case x:
+  case z:
+    return true;
+  default:
+    doSomething();
+}
+~~~~~~~~~~~~~
+
+================================================================================
+\section code_guidelines_5 Naming
+\subsection code_guidelines_5_1 Namespaces
+
+Namespaces should be in uppercase letters
+
+~~~~~~~~~~~~~
+namespace KODI
+{
+...
+}
+~~~~~~~~~~~~~
+
+\subsection code_guidelines_5_2 Constants
+
+Use upper case with underscore spacing where necessary.
+
+~~~~~~~~~~~~~
+const int MY_CONSTANT = 1;
+~~~~~~~~~~~~~
+
+--------------------------------------------------------------------------------
+\subsection code_guidelines_5_3 Enums
+
+Use CamelCase for the enum name and upper case for the values.
+
+~~~~~~~~~~~~~
+enum Dummy
+{
+  VALUE_X,
+  VALUE_Y
+};
+~~~~~~~~~~~~~
+
+--------------------------------------------------------------------------------
+\subsection code_guidelines_5_4 Interfaces
+
+We use CamelCase for interface names and they should be prefixed with an
+uppercase I. Filename should match the interface name, e.g. `ILogger.h`
+
+~~~~~~~~~~~~~
+class ILogger
+{
+  void Log(...) = 0;
+}
+~~~~~~~~~~~~~
+
+--------------------------------------------------------------------------------
+\subsection code_guidelines_5_5 Classes
+
+We use CamelCase for class names and they should be prefixed with an uppercase C.
+Filename should match the class name without the prefixed C, e.g. `Logger.cpp`
+
+~~~~~~~~~~~~~
+class CLogger : public ILogger
+{
+  void Log(...)
+}
+~~~~~~~~~~~~~
+
+--------------------------------------------------------------------------------
+\subsection code_guidelines_5_6 Methods
+
+We use CamelCase for method names and first letter should always be upper case.
+Even if the methods are private or protected.
+
+~~~~~~~~~~~~~
+void MyDummyClass::DoSomething();
+~~~~~~~~~~~~~
+
+--------------------------------------------------------------------------------
+\subsection code_guidelines_5_7 Variables
+
+We use CamelCase for variables. Type prefixing is optional.
+\subsubsection code_guidelines_5_7_1 Global Variables
+
+Prefix global variables with g_
+
+~~~~~~~~~~~~~
+int g_globalVariableA;
```

Comment:
Better to remove complete?

For my is this also a bad example, to use a editable integer as global is a worst way to use.
Only for me a safe example can be a class with thread safe functions.